2025 Hyperledger Web3j Annual Review

gtebrean opened the pull request.

tkuhrt commented:
@mbrandenburger (primary) and @cjhowland (secondary)

mbrandenburger commented:
Thank you @gtebrean for submitting your review report. @cjhowland and I had a look and post here our summary. We would like to discuss your report in the [next TAC](https://github.com/tkuhrt/lfdt-governance/blob/20230320/tac/meeting-minutes/2025/2025-03-20.md) and invite you to join.

# Review Summary

## Project Health

The Web3j project has an active community; however, the number of active contributors has been declining since 2022. May 2023 was the strongest month for contributions, but activity has decreased since then.

Initially, the project was called Hyperledger Web3j, but dropped the Hyperledger prefix to minimize confusion with existing users as stated on Discord.

The project maintains 14 repositories and continues to actively release new versions. The latest release is v4.13.0. Notably, there has been good progress on the roadmap over the past six months.

The project maintains an active Discord channel, and multiple community events have taken place in 2024, including a workshop on Web3j and various meetups. 

Project adoption is postive. At least 12 organizations are currently using Web3j for production use cases.


## Maintainer Diversity

There are currently only two active maintainers, both affiliated with Web3Labs. Four maintainers are listed via the MAINTAINERS file, but two appear to qualify for Emeritus status due to inactivity.


## Goals

A major release is planned:

Web3j 5.x – targeted for Q3 2025


## Lifecycle

The project's Incubation status remains appropriate.


## Suggestions for Improvement

- Work on increasing maintainer diversity.

- Introduce release artifacts for the planned.

- Address minor inconsistencies in the roadmap (e.g., reducing issues by 80% vs. 40%).

- Integrate OpenSSF Scorecard to identify necessary improvements torwards project graduation.


## Open Questions

- Why was activity between March and July 2023 significantly higher than in later months (e.g., October onward)?

- "The MAINTAINERS.md file is up-to-date." As mentioned above, two maintainers appear to be inactive.

- Three virtual events were held to promote the project and attract maintainers. Would you like to list them here? (Example: LFDT London Meetup, workshop on YouTube (March 2024), )

- The project is open to mentoring new contributors and maintainers. What specific steps are being taken to support this? Are there easy-to-fix issues or low-hanging fruit contributions?

- Have other LFDT projects been consulted regarding best practices for adoption, contributions, and onboarding?

- Would you like to mention the Web3j mentorship project in this report?

- Are there any ongoing efforts to support workshops and summits?

gtebrean commented:
@mbrandenburger thanks for the review! Here are the answers for your questions:

Why was activity between March and July 2023 significantly higher than in later months (e.g., October onward)?
- I took a look and there were only 3 PRs merged in that period, so I don't think it was significantly higher. As 95% of the activities are done by Web3 Labs, period depends be the other projects workloads and priorities as web3j is open source. 

"The MAINTAINERS.md file is up-to-date." As mentioned above, two maintainers appear to be inactive.
- Updated, 3 active maintainers

Three virtual events were held to promote the project and attract maintainers. Would you like to list them here? (Example: LFDT London Meetup, workshop on YouTube (March 2024), )
- added

The project is open to mentoring new contributors and maintainers. What specific steps are being taken to support this? Are there easy-to-fix issues or low-hanging fruit contributions?
- updated in the maintainers section

Have other LFDT projects been consulted regarding best practices for adoption, contributions, and onboarding?
- Yes, we are consulting Besu for which we use to contribute and Indy

Would you like to mention the Web3j mentorship project in this report?
- added

Are there any ongoing efforts to support workshops and summits?
- yes, we are working on features in order to have topics to present

mbrandenburger commented on tac/project-updates/2025/2025-annual-Web3j.md:
```diff
@@ -0,0 +1,77 @@
+
+## Maintainer Diversity
+
+At the moment there are only 3 maintainers (both from Web3 Labs).
```
```suggestion
At the moment there are only 3 maintainers (all from Web3 Labs).
```

ryjones commented:
Approved 03 APR 2025, eight aye, three absent
